src/main/webapp/js/maps-page-loader.js

```diff
@@ -112,6 +112,7 @@ async function handleSumbitButtonClick(map, lat, lng, cinemaName, key, marker, t
   const delay = ms => new Promise(res => setTimeout(res, ms));
   await postMessage(key, text);
   marker.setMap(null);
+  //set delay to ensure message is persisted in data store and can be retrieved to be shown in the info window
   await delay(4500);
```

Comment:
I'm not sure how the 4500 (4.5 seconds i assume) value was obtained, but I think it would be better to code it in a way such that => x happens after y happens (like execute x only when y is executed)? Is something like that possible? Because I am a little doubtful of using a hard coded time value...is this value dependent on internet speed or has some dependent variables? Because what happens if my internet speed is slower than average and for me I need a longer time delay than 4.5secs? 

Would sync functions work better?

Reply:
Also, I personally think between latency for the user of our webpage and the length of the function, its more important to shorten the latency for the users as much as possible? Because I think a more than double increase in latency  @ 4.5s is a little too much and if there's a very direct way to solve this problem (having a longer function) then I think it'll be better to make it 2s instead?